streams/src/main/java/org/apache/kafka/streams/state/internals/TimestampedSegments.java

```diff
@@ -35,7 +40,8 @@ public TimestampedSegment getOrCreateSegment(final long segmentId,
         if (segments.containsKey(segmentId)) {
             return segments.get(segmentId);
         } else {
-            final TimestampedSegment newSegment = new TimestampedSegment(segmentName(segmentId), name, segmentId);
+            final TimestampedSegment newSegment =
+                new TimestampedSegment(segmentName(segmentId), name, segmentId, metricsRecorder);
```

Comment:
The same recorder instance is passed to the RocksDB store instance of each segment.